contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/EditCategoriesModal.vue

```diff
@@ -0,0 +1,330 @@
+         * {
+         *  [categoryId]: true | [nodeId1, nodeId2, ...]
+         * }
+         * If the value is true, it means that the category is selected for all nodes
+         * If the value is an array of nodeIds, it means that the category is selected
+         * just for those nodes
+         */
+        selectedCategories: {},
+      };
+    },
+    computed: {
+      ...mapGetters('contentNode', ['getContentNodes']),
+      nodes() {
+        return this.getContentNodes(this.nodeIds);
+      },
+      isTopicSelected() {
+        return this.nodes.some(node => node.kind === ContentKindsNames.TOPIC);
+      },
+      categoriesOptions() {
+        const searchQuery = this.searchQuery?.trim().toLowerCase();
+        if (!this.searchQuery) {
+          return this.categories;
+        }
+        return this.categories.filter(category =>
+          category.label.toLowerCase().includes(searchQuery)
+        );
+      },
+      flatList() {
+        return this.searchQuery && this.searchQuery.trim().length > 0;
+      },
+      dividerStyle() {
+        return {
+          border: 0,
+          borderBottom: `1px solid ${this.$themeTokens.fineLine}`,
+          margin: '1em 0',
+        };
+      },
+      categoriesWithMixedOptions() {
+        return [
+          {
+            value: MIXED,
+            label: this.$tr('mixedLabel'),
+            undeletable: true,
+          },
+          ...this.categories,
+        ];
+      },
+      selectedCategoriesValues() {
+        const selectedCategories = Object.entries(this.selectedCategories)
+          .filter(entry => entry[1] === true) // no mixed values
+          .map(([key]) => key);
+        if (Object.values(this.selectedCategories).some(value => value !== true)) {
+          selectedCategories.push(MIXED);
+        }
+        return selectedCategories;
+      },
+    },
+    created() {
+      const categories = getSortedCategories();
+      this.categories = Object.entries(categories).map(([id, category]) => ({
+        value: id,
+        label: this.translateMetadataString(camelCase(category)),
+        level: id.split('.').length,
+      }));
+
+      const categoriesNodes = {};
+
+      this.nodes.forEach(node => {
+        Object.entries(node.categories || {})
+          .filter(entry => entry[1] === true)
+          .forEach(([category]) => {
+            categoriesNodes[category] = categoriesNodes[category] || [];
+            categoriesNodes[category].push(node.id);
+          });
+      });
+
+      Object.entries(categoriesNodes).forEach(([key, nodeIds]) => {
+        if (nodeIds.length === this.nodeIds.length) {
+          this.selectedCategories[key] = true;
+        } else {
+          this.selectedCategories[key] = nodeIds;
+        }
+      });
+    },
+    methods: {
+      ...mapActions('contentNode', ['updateContentNode', 'updateContentNodeDescendants']),
+      close() {
+        this.$emit('close');
+      },
+      isCheckboxSelected(category) {
```

Comment:
Could we restructure this method some more(You might need to move out some pieces into properly named methods). Even with the comments, its not very clear what is happening.